### PR TITLE
Inline variant definition in the documentation after #15484.

### PR DESCRIPTION
**Pull request by Zimmi48**
If someone is motivated, it would be really nice to improve the description of `Variant` and `Record`. The former should be presented directly, not primarily as a restricted `Inductive`, because we have put the section about variants and the match construct before the one of inductive types and fixpoints.

### CODE REVIEW

**Comment by Zimmi48** [comment]
Oh actually, the part on records was already done in #15489.

**Comment by jfehrle** [comment]
@Zimmi48 What do you want to do with this one?  I hadn't noticed the PR before now.

**Comment by Zimmi48** [comment]
I will rebase after merging Pierre's PR.

**Comment by Zimmi48** [comment]
Rebased.

**Comment by ppedrot** [comment]
What is blocking this?

**Comment by ppedrot** [comment]
Postponing.

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@Zimmi48 Can you confirm that this is good to go. We can open an issue for the improvement of this part of the documentation if you like, but it doesn't seem part of this PR.

**Comment by Zimmi48** [comment]
Yes, this is good to go (when it isn't, I put the PR as draft or I add a label to say so). Thanks!

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**
